codemon/CodemonMeta.py

```diff
@@ -34,3 +38,52 @@ def get_filename():
 def get_practice_files():
   practiceFiles = ['A.cpp','B.cpp','C.cpp']
   return practiceFiles
+
+def write_to_file(filename, text, contestName=None):
+  full_filename = os.path.join(os.getcwd(), filename)
+  if contestName is not None:
+    full_filename = os.path.join(os.getcwd(), contestName, filename)
+  with open(full_filename, 'w+') as f:
+    f.write(text)
+
+def compile_and_run(filename):
+  # Store full file paths
+  full_filename = os.path.join(os.getcwd(), filename)
+  full_output_filename = os.path.join(os.getcwd(), 'prog')
+  full_input_filename = os.path.join(os.getcwd(), 'input.txt')
+
+  # Check if required files exist
+  if not Path(full_filename).is_file():
+    print(f"{filename} doesn't exist.", file=sys.stderr)
+  if not Path(full_input_filename).is_file():
+    print(f"'input.txt' doesn't exist.", file=sys.stderr)
+
+  compilation_child_process = subprocess.Popen(['g++', full_filename, '-o', full_output_filename], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+  compilation_child_process.wait()
+  compilation_child_process.terminate()
+  if compilation_child_process.returncode != 0:
+    return_code = compilation_child_process.returncode
+    if return_code == 127:
+      print("'g++' isn't installed.", file=sys.stderr)
+    else:
+      print('Error in compilation.', file=sys.stderr)
+      print(f'Return code: {return_code}', file=sys.stderr)
+    return
+  print('Running...')
+  execution_child_process = subprocess.Popen([full_output_filename], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+  input_text = ''
+  with open(full_input_filename, 'r+', encoding='utf-8') as infile:
+    input_text = infile.read()
+  if len(input_text) > 0 and not input_text.isspace():
+    print(colored.yellow('Taking input from input.txt.'))
+    execution_child_process.stdin.write(input_text.encode(encoding='utf-8'))
+  else:
+    print(colored.yellow('input.txt is empty.\nSkipping input.'))
+  output = execution_child_process.communicate()
+  if len(output) > 0:
+    print(colored.green(output[0].decode()))
+    print()
+  else:
+    print(colored.red('No output.'))
+  execution_child_process.terminate()
+  print(f'Programme returned {execution_child_process.returncode}.')
```

Comment:
I like the refactoring you've done but I think these functions should be in the `CodemonInit.py` file. Could you shift them there ?
Re: `write_to_file` should be shifted to `CodemonInit.py` and `compile_and_run` should be shifted to `CodemonListen.py`.